addons/account/report/account_invoice_report.py

```diff
@@ -99,15 +97,41 @@ def _select(self):
                 move.invoice_partner_bank_id,
-                -line.balance * (move.amount_residual_signed / NULLIF(move.amount_total_signed, 0.0)) * (line.price_total / NULLIF(line.price_subtotal, 0.0))
-                                                                            AS residual,
-                -line.balance * (line.price_total / NULLIF(line.price_subtotal, 0.0))    AS amount_total,
+                ROUND(move.amount_residual_signed / (SELECT count(*) FROM account_move_line aml where line.move_id = aml.move_id AND NOT aml.exclude_from_invoice_tab),
+                  COALESCE((SELECT decimal_places
+                    FROM res_currency rc INNER JOIN res_currency_rate cr ON
+                    rc.id = cr.currency_id
+                   WHERE cr.currency_id = (COALESCE(line.currency_id, line.company_currency_id)) AND
+                         cr.company_id = line.company_id
+                   LIMIT 1
+                   ),2))                                                   AS residual,
+                ROUND(
+                  line.price_total / COALESCE(
+                    (SELECT rate FROM res_currency_rate cr WHERE
+                      cr.currency_id = line.currency_id AND
+                      cr.company_id = line.company_id AND
+                      cr.name <= COALESCE(line.date,NOW())
+                    ORDER BY cr.name DESC
+                    LIMIT 1)
+                    ,1),
+                  COALESCE((SELECT decimal_places
+                   FROM res_currency rc INNER JOIN res_currency_rate cr ON
+                    rc.id = cr.currency_id
+                   WHERE cr.currency_id = (COALESCE(line.currency_id, line.company_currency_id)) AND
+                         cr.company_id = line.company_id
+                   LIMIT 1
+                   ),2))
+                    *
+                   (CASE WHEN move.amount_total_signed < 0
+                        THEN -1
+                        ELSE 1
+                    END)                                                    AS amount_total,
                 uom_template.id                                             AS product_uom_id,
                 template.categ_id                                           AS product_categ_id,
-                line.quantity / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0)
+                SUM(line.quantity / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0))
                                                                             AS quantity,
-                -line.balance                                               AS price_subtotal,
-                -line.balance / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0)
+                -SUM(line.balance)                                          AS price_subtotal,
+                -SUM(line.balance / NULLIF(COALESCE(uom_line.factor, 1) * COALESCE(uom_template.factor, 1), 0.0))
```

Comment:
I'm not sure the reverse is the good option, but this operator change is not something you want
Same for the SUM function, that's not something you cant in the query.